app/services/hyrax/file_set_characterizer.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Characterizes an ActiveFedora based FileSet
+  class FileSetCharacterizer
+    ##
+    # @!attribute [rw] source
+    #   @return [#characterize]
+    attr_accessor :source
+
+    ##
+    # @param source the object to characterize
+    def initialize(source:)
+      self.source = source
```

Comment:
Should this be `@source = source`?

Reply:
I personally prefer using a setter method, as it allows checks/coercion.